equinix/resource_network_device.go

```diff
@@ -1467,7 +1478,7 @@ func getNetworkDeviceStateChangeConfigs(c ne.Client, deviceID string, timeout ti
 	configs := make([]*resource.StateChangeConf, 0, len(changes))
 	for change, changeValue := range changes {
```

Comment:
Should we be using a `for` loop here? We know that we are only concerned with three fields: `MgmtAclTemplateUuid`, `ACLTemplateUUID`, and `AdditionalBandwidth`.
 
Per our conversation, it sounds like there is a dependency that AdditionalBandwidth can only be applied if an ACL is defined. Therefor, we should be applying these changes in a predefined order, rather than the order the random order the `changes` map is iterated in.

Reply:
something like: 

```go
if change, found := changes[networkDeviceSchemaNames["ACLTemplateUUID"]]; found {
...
}
if change, found := changes[networkDeviceSchemaNames["MgmtAclTemplateUuid"]]; found {
...
}
if change, found := changes[networkDeviceSchemaNames["AdditionalBandwidth"]]; found {
...
}
```

Reply:
This would process the fields in order, assuring that an ACL is applied before the Bandwidth is applied.

Reply:
addressed in 0952637